src/test/groovy/graphql/LocalContextAwareCascadingTest.groovy

```diff
@@ -0,0 +1,122 @@
+                    .type(TypeRuntimeWiring.newTypeWiring("Blog")
+                            .dataFetcher("comments", {
+                                def comment = new Comment()
+                                comment.title = it.getLocalContext() + " (comments data fetcher)"
+                                DataFetcherResult.newResult()
+                                        .data([comment])
+                                        .build()
+                            })
+                    )
+                    .type(TypeRuntimeWiring.newTypeWiring("Comment")
+                            .dataFetcher("body", { it.getLocalContext() + " (comment data fetcher)" })
+                            .dataFetcher("author", { new User() })
+                    )
+                    .type(TypeRuntimeWiring.newTypeWiring("User")
+                            .dataFetcher("username", { it.getLocalContext() + " (user data fetcher)" })
+                    )
+                    .build())
+            .build()
+
+    def "when a local context is provided, it can be consumed by the grandchildren node data fetchers"() {
+        given:
+            def input = ExecutionInput.newExecutionInput()
+                    .query('''
+                            query {
+                                blog {
+                                    name
+                                    comments {
+                                        title
+                                        body
+                                        author {
+                                            username
+                                        }
+                                    }
+                                }
+                            }
+                            ''')
+                    .build()
+        when:
+            def executionResult = graphql.execute(input)
+
+        then:
+            executionResult.errors.isEmpty()
+            executionResult.data == [
+                    blog: [
+                            name    : "some name",
+                            comments: [
+                                    [
+                                            title: "BLOG CONTEXT (comments data fetcher)",
+                                            body : "BLOG CONTEXT (comment data fetcher)",
+                                            author: [
+                                                    username: "BLOG CONTEXT (user data fetcher)"
+                                            ]
+                                    ]
+                            ]
+                    ]
+            ]
+    }
+
+    // Really simply beans, without getters and setters for simplicity
+    static class User {
+        public String username
+    }
+
+    static class Comment {
+        String title
+        String body
+        User author
+    }
+
+    static class Blog {
+        String name
+        List<Comment> comments
+    }
+}
```

Comment:
FYI - The PropertyDataFetcher also handles maps (via field names)

So you can use POJOs like above or Maps if that easier in Groovy

```
   def blog  = [ name : "some name", comments : [ [ title : "comment title", body : "body text", author : [...] ] ] 
```

not saying you can use POJOs just FYI you can use maps.

Do what reads well!

Reply:
Ah yeah that probably would have been easier with the maps